onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1881,19 +1908,38 @@ Status ConcatOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const
   input_names.reserve(node_input_size);
   for (size_t i = 0; i < node_input_size; i++) {
     const auto& input = inputs[i].node_arg.Name();
+
+    if (is_quant_op) {
+      float scale = 0.0f;
+      int32_t zero_point = 0;
+      ORT_RETURN_IF_ERROR(GetQuantizationScaleAndZeroPoint(
+          model_builder.GetInitializerTensors(), node_unit.Inputs()[i], node_unit.ModelPath(),
+          scale, zero_point));
+
+      ORT_RETURN_IF_ERROR(IsValidInputQuantizedType(model_builder, input, scale, zero_point));
+    }
```

Comment:
Is no check that scale and zp is consistent required here?

Reply:
No, for quantized op, this is already checked in the OpSupportChecker